### ISSUE TITLE
Custom Toolbox Tab OnSelected is not called after a recompile of the plugin

### ISSUE THREAD

**Issue by schmidt-florian** [FlaxEngine/FlaxEngine#1843]
**Issue description:**

I have following Code:

```csharp
public class GameEditorPlugin : EditorPlugin
{
    private TileTab _tileTab;
    private Editor _editor;
    
    public override void Initialize()
    {
        base.Initialize();
        
        _editor = Editor.Instance;
        _editor.Windows.ToolboxWin.TabsControl.AddTab(_tileTab = new TileTab(Editor.Icons.Grid32, Editor));
    }

    public override void Deinitialize()
    {
        base.Deinitialize();
        _tileTab.Dispose();        
    }
}

public class TileTab : Tab
{
    private readonly Editor _editor;

    public TileTab(SpriteHandle icon, Editor editor)
        : base(string.Empty, icon)
    {
        _editor = editor;
    }

    /// <inheritdoc />
    public override void OnSelected()
    {
        base.OnSelected();
        Debug.Log("Selected");
    }

    public override void OnDeselected()
    {
        base.OnDeselected();
        Debug.Log("Deselected");        
    }
}
```

The issue is, if i have my Tab in the toolbox selected and recompile the code, my tab stays selected in the Editor Toolbox but the `OnSelected` is not called.

If I switch tabs `OnSelected` is called as expected but as I said, not on recompile when it is selected.

**Steps to reproduce:**
1. Add the code to a project.
2. Select the tab -> View Log
3. Hit recompile
4. The Tab will stay selected in the UI but it will miss the `OnSelected`

**Flax version:**
master 37a6ea9

**Comment by mafiesto4** [comment]
FYI: Plugins are initialized before Editor startup so if you use Editor customizations use `InitializeEditor` method:

```cs
public class GameEditorPlugin : EditorPlugin
{
    private TileTab _tileTab;

    public override void InitializeEditor()
    {
        base.InitializeEditor();

        Editor.Windows.ToolboxWin.TabsControl.AddTab(_tileTab = new TileTab(Editor.Icons.Grid32, Editor));
    }

    public override void DeinitializeEditor()
    {
        _tileTab.Dispose();

        base.DeinitializeEditor();
    }
}
```

**Comment by mafiesto4** [comment]
Fixed in c78ce9697cb3acacd525bed472d7b318ebf8e835.
